tuplex/core/src/ee/local/LocalBackend.cc

```diff
@@ -105,29 +105,27 @@ namespace tuplex {
         assert(stage);
 
         // reset history server
-        _historyServer.reset();
+//        _historyServer.reset();
 
         if(!stage)
             return;
 
         // history server connection should be established
         bool useWebUI = _options.USE_WEBUI();
         // register new job
-        if(useWebUI) {
+        if(useWebUI && stage->predecessors().size() == 0) {
```

Comment:
please add a comment on what this checks. Consider changing size() == 0 to empty() function.

Reply:
Added comment. Changed to empty().